src/docgen.cpp

```diff
@@ -23,15 +24,25 @@ static inline void set_options(int argc, char **argv)
 	while ((c = getopt(argc, argv, ":i:o:d:")) != -1) {
 		switch (c) {
 			case 'i':
-				parsed_src = strcmp(optarg, "-") == 0 ? &std::cin : new std::ifstream(optarg);
-				if (parsed_src->fail()) {
-					throw docgen::system_error("failed to open istream");
+				if (strcmp(optarg, "-") == 0) {
```

Comment:
This code is exactly the same as the one below, can we make a separate helper function that takes care of both cases? This kind of helper function can be placed in a `namespace details` very C++y ;)

Reply:
It's similar, but this exact logic will never be needed outside these two examples—I understand the inclination, but I don't think it's worth making the sacrifice to readability for abstraction in this case.

Reply:
ok later if we add more flags or something and this is needed we'll come back to it. LGTM